app/models/goal.py

```diff
@@ -1,6 +1,21 @@
 from flask import current_app
 from app import db
 
-
 class Goal(db.Model):
     goal_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.Text)
+    tasks = db.relationship("Task", backref="goals", lazy=True)
+    __tablename__ = "goals"
+    
+    def to_json(self):
+        if self.tasks != []:
```

Comment:
This can be simplified as:

```suggestion
        if self.tasks:
```

Since empty lists are falsey.